### PR TITLE
Feature/versioned encryption

### PR DESCRIPTION
**Pull request by cm421**
### Intent

Addresses https://github.com/rstudio/rstudio-pro/issues/6412 by adding versioned encryption functions.

Also:

- Sets default encryption to Version 0 (which is what it's always been. Now it just has a name)
- Adds unit tests around each encryption version

### Approach

This PR just introduces versions and sets us to use Version 0. Future PRs will add logic to use them.

### Automated Tests

Unit tests updated to include tests for each version of encryption

### QA Notes

Expectation is that no current tests should fail. We're using the same encryption code, it just now has a new name and a new home. The new unit tests should cover the added encryption versions

### Documentation

None yet until feature is complete.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by jeffvroom — APPROVED** [review]
Nice!  Super clean and easy to read.

**Review by dmortondev — APPROVED** [review]
LGTM! Like Jeff said, very clean and easy to read solution!

Given the API hasn't changed Launcher shouldn't have a hard time picking this up. I'll start that process once this is merged and promoted to the pro repo.

**Comment by cm421** [comment]
Thanks @dmortondev! Just as a heads up, there will be a series of PRs until the feature is complete, so unless it's easier to multiple smaller merges you can hold off on merging anything specifically for these changes until all the future PRs are in too

**Pull request merged**
